### PR TITLE
dts: qcom: msm8953-xiaomi-vince: add hall sensor and update config fragment

### PR DESCRIPTION
**Pull request by M0Rf30**
To test the hall sensor:
- fire up evtest on /dev/input/event0 
- use a magnet and move it  on the back of the phone, near to the silver line on the bottom (where the model is written, to be more specific)

```
Event: time 1682501052.376060, type 5 (EV_SW), code 0 (SW_LID), value 1
Event: time 1682501052.376060, -------------- SYN_REPORT ------------
Event: time 1682501052.465216, type 5 (EV_SW), code 0 (SW_LID), value 0
Event: time 1682501052.465216, -------------- SYN_REPORT ------------
```

### CODE REVIEW

**Comment by M0Rf30** [comment]
@fekz115 @alikates @z3ntu

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-vince.dts [hunk 1]
```diff
@@ -17,6 +17,21 @@
 	compatible = "xiaomi,vince", "qcom,msm8953";
 	qcom,board-id= <0x1000b 0x08>;
 
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_key_default>, <&gpio_hall_sensor_default>;
```
Comment (z3ntu) [inline arch/arm64/boot/dts/qcom/msm8953-xiaomi-vince.dts:24]: You're somewhat weirdly overwriting/extending the gpio-keys node from xiaomi-common.dtsi, so logically `gpio_key_default` doesn't fit here at all also.

**Comment by z3ntu** [comment]
Please send features like this directly upstream also, since it doesn't have any driver dependency.

File: arch/arm64/configs/xiaomi-vince.config [hunk 2]
```diff
@@ -1,9 +1,17 @@
+CONFIG_AK09911=m
```
Comment (JIaxyga) [inline arch/arm64/configs/xiaomi-vince.config:1]: not CONFIG_AK09911. CONFIG_AK8975

**Pull request merged**